jetbrains-core/tst/software/aws/toolkits/jetbrains/services/s3/CreateS3BucketDialogTest.kt

```diff
@@ -0,0 +1,98 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.s3
+
+import com.intellij.testFramework.ProjectRule
+import com.intellij.testFramework.runInEdtAndWait
+import com.nhaarman.mockitokotlin2.argumentCaptor
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.doThrow
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.reset
+import com.nhaarman.mockitokotlin2.stub
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import software.amazon.awssdk.services.s3.S3Client
+import software.amazon.awssdk.services.s3.model.BucketAlreadyExistsException
+import software.amazon.awssdk.services.s3.model.CreateBucketRequest
+import software.amazon.awssdk.services.s3.model.CreateBucketResponse
+import software.aws.toolkits.jetbrains.utils.DelegateSdkConsumers
+
+class CreateS3BucketDialogTest {
+    @Rule
+    @JvmField
+    val projectRule = ProjectRule()
+
+    private val s3Mock = mock<S3Client>(defaultAnswer = DelegateSdkConsumers())
+
+    @Before
+    fun setUp() {
+        reset(s3Mock)
```

Comment:
Not sure if this is required, I thought Mockito/JUnit reset the class class for each test method

Reply:
Nice catch. Removed.

Reply:
The reset will only happen if you're using the `@RunWith(JUnitMockitoRunner)` which we ain't.